crates/cext/README.md

```diff
@@ -0,0 +1,60 @@
+# `qiskit-cext`
+
+This crate contains the bindings for Qiskit's C API. 
+
+## Building the library
+
+The C bindings are compiled into a shared library, which can be built along with the header file
+by running
+```
+$ make cheader
+```
+in the root of the repository. The header file, `qiskit.h`, is generated using 
+[cbindgen](https://github.com/mozilla/cbindgen) and stored in `dist/c/include`.
+
+The following example uses the header to build a 100-qubit observable:
+```c
+#include <stdio.h>
+#include <qiskit.h>
+
+int main(int argc, char* argv[]) {
+// build a 100-qubit empty observable
+u_int32_t num_qubits = 100;
+QkSparseObservable *obs = qk_obs_zero(num_qubits);  
+
+// add the term 2 * (X0 Y1 Z2) to the observable
+complex double coeff = 2;  // the coefficient
+QkBitTerm bit_terms[3] = {QkBitTerm_X, QkBitTerm_Y, QkBitTerm_Z};  // bit terms: X Y Z
+uint32_t indices[3] = {0, 1, 2};  // indices: 0 1 2
+QkSparseTerm term = {&coeff, 3, bit_terms, indices, num_qubits};
+qk_obs_add_term(obs, &term);  // append the term
+
+// print some properties
+printf("num_qubits: %i\n", qk_obs_num_qubits(obs));
+printf("num_terms: %i\n", qk_obs_num_terms(obs));
+
+// free the memory allocated for the observable
+qk_obs_free(obs);
+
+return 0;
+}
+```
+Refer to the C API documentation for more information and examples.
+
+## Compiling
+
+The above program can be compiled by including the header and linking to the `qiskit_cext` library,
+which is located in `target/release`. (The exact name depends on the platform, e.g.,
+`target/release/libqiskit_cext.dylib` on MacOS.) 
```

Comment:
We can follow up on this, but we should have a Makefile command that builds `libqiskit_cext.so` and then copies it into `dist/c/lib` with the name `libqiskit.{so,dylib,dll}` as appropriate.  The "cext" name is an internal implementation detail, and we should be building the object into the distribution location, along with the header file.

Reply:
Let's do this in a follow up 👍🏻

Reply:
I did this in 6869d1b4bf, so the usage instructions can all be `-lqiskit` etc.